packages/design-system/src/molecules/selectV2/index.tsx

```diff
@@ -151,6 +161,77 @@ export const SelectV2: React.FC<SelectV2Props> = ({
     [renderReference, selectedValue],
   )
 
+  const renderOptions = useCallback(() => {
+    return filteredOptions.map((option, index) => (
+      <Pressable
+        key={option.value}
+        onPress={() => {
+          setValue(option.value)
+          onChange?.(option.value)
+          handleOptionPress?.()
+        }}
+        _hover={{
+          style: [{ backgroundColor: theme.base.highlight2[10] }],
+        }}
+        _focusVisible={{
+          style: [{ backgroundColor: theme.base.highlight2[10] }],
+        }}
+        testID={`selectOption__${option.value}`}
+      >
+        <TypographyV2
+          variant="caption1"
+          color={theme.text.black[100]}
+          style={[padding.ph24, padding.pv12]}
+        >
+          {option.name}
+        </TypographyV2>
+        {index !== options.length - 1 && (
+          <View
+            style={[
+              styles.divider,
+              { backgroundColor: theme.background.silver },
+            ]}
+          />
+        )}
+      </Pressable>
+    ))
+  }, [filteredOptions, onChange, handleOptionPress, options, theme])
+
+  const renderNoOptions = useCallback(
+    () => (
+      <View>
+        <TypographyV2
+          variant="caption1"
+          style={[padding.ph24, padding.pv12, text.center]}
+        >
+          There are no options available
+        </TypographyV2>
+      </View>
+    ),
+    [],
+  )
+
+  const renderLoading = useCallback(
```

Comment:
Great idea to separate the renderOptions, renderNoOptions and renderLoading into separate functions